modules/core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/resolver/TestDocumentModelResolver.java

```diff
@@ -671,6 +681,32 @@ public void testUserWithNoAccessToDocument() {
         }
     }
 
+    // NXP-30192
+    @Test
+    public void testFetchWithBrowseAndReadPermission() {
+        // document bob can only browse
+        DocumentModel barDoc = session.createDocumentModel("/", "bar", "TestResolver");
+        barDoc = session.createDocument(barDoc);
+        ACP acp = barDoc.getACP();
+        acp.addACE(ACL.LOCAL_ACL, new ACE("bob", "Browse", true));
+        barDoc.setACP(acp, true);
+        // document bob can read
+        DocumentModel fooDoc = session.createDocumentModel("/", "foo", "TestResolver");
+        fooDoc.setPropertyValue(REPO_AND_ID_XPATH, barDoc.getRepositoryName() + ":" + barDoc.getId());
+        fooDoc = session.createDocument(fooDoc);
+        acp = fooDoc.getACP();
+        acp.addACE(ACL.LOCAL_ACL, new ACE("bob", "Read", true));
+        fooDoc.setACP(acp, true);
+        transactionalFeature.nextTransaction();
+
+        CoreSession bobSession = coreFeature.getCoreSession("bob");
+        DocumentModel resolvedDoc = (DocumentModel) fooDoc.getProperty(REPO_AND_ID_XPATH)
+                                                          .getObjectResolver()
+                                                          .fetch(bobSession);
+        // bob cannot read the referenced document
+        assertNull(resolvedDoc);
```

Comment:
Could you also test explicitly what happens on a list? I guess here the list will contain a `null`, but as we discussed another option would be to skip the value from the list completely.